### PR TITLE
Add a `@mermaidchart/cli` CLI tool for accessing Mermaid Chart

### PR DESCRIPTION
**Pull request by aloisklink**
## 📖 Description

> [!Note]
>
> This PR is stacked on top of:
> - #17

Add a @mermaidchart/cli package that can `link`/`pull`/`push` diagrams, e.g.

```console
me@pc:~$ npx mermaid-chart link ./my-diagram.mmd 
? Select a project to upload your document to personal
me@pc:~$ npx mermaid-chart push ./my-diagram.mmd
✅ - ./my-diagram.mmd  was pushed
me@pc:~$ npx mermaid-chart pull ./my-diagram.mmd  
✅ - ./my-diagram.mmd  is up to date
```

```console
me@pc:~$ npx @mermaidchart/cli --help
Usage: mermaid-cli [options] [command]

CLI for interacting with https://MermaidChart.com, the platform that makes
collaborating with Mermaid diagrams easy.

Options:
  -V, --version               output the version number
  -c, --config <config_file>  The path to the config file to use. (default:
                              "/home/me/.config/mermaid-chart.toml")
  -h, --help                  display help for command

Commands:
  whoami                      Display Mermaid Chart username
  login                       Login to a Mermaid Chart account
  logout                      Log out of a Mermaid Chart account
  link <path>                 Link the given Mermaid diagram to Mermaid Chart
  pull [options] <path>       Pulls a document from from Mermaid Chart
  push <path>                 Push a local diagram to Mermaid Chart
  help [command]              display help for command
```

### Configuration file

The configuration file is a [TOML file](https://toml.io/en/) that's stored in:

- Linux: `~/.config/mermaid-config.toml`
- MacOS: `~/Library/Application Support/mermaid-config.toml`
- Windows: `%AppData%\mermaid-config.toml`

Currently, the allowed fields are:

```toml
# TOML config file for mermaid-chart
[mermaid-chart]
base_url = "https://test.mermaidchart.com" # optional, defaults to https://mermaidchart.com
auth_token = "xxxxxx-xxxxxxx-xxxxxxxx" # set by `mermaid-chart login`.
```

### Todo

- [x] ~ESLint is currently broken, I think due to the version of TypeScript I'm using.~
  - Fixed: I needed to install `@typescript-eslint/parser` v6.
- [ ] Add OAuth login. Currently, users have to manually go to https://www.mermaidchart.com/app/user/settings to generate a key.
- [x] Currently, we can only `pull` or `push` a single diagram at a time. It would be better to support multiple diagrams at once (e.g. using a glob).
  - Will be implemented by PR #15
- [ ] Test on MacOS/Windows.
  - MacOS/Windows are now tested in CI: https://github.com/Mermaid-Chart/plugins/pull/14, which helped find the bug in https://github.com/Mermaid-Chart/plugins/pull/12/commits/f234f3ec22e98c1d7f8ef712769fc3a648c92bea. We still need manual/usability tests, though.
- [ ] Support keeping comments in YAML frontmatter, as they're currently all removed.
  - Will be fixed by #21
- [ ] Publish to NPM.
  - [ ] We first need to make a new release of [`@mermaidchart/sdk`](https://www.npmjs.com/package/@mermaidchart/sdk).

### CODE REVIEW

File: packages/cli/package.json
Comment (aloisklink) [inline packages/cli/package.json:30]: @knsv, what do you want the license for this package to be?

MIT is the license in the root of the repo, so we could use that: https://github.com/Mermaid-Chart/plugins/blob/main/LICENSE

File: packages/cli/test/fixtures/connected-diagram.mmd
Comment (aloisklink) [inline packages/cli/test/fixtures/connected-diagram.mmd:3]: I'm slightly leaning towards having a URL here instead of an ID, e.g. something like:

```yaml
id: https://mermaidchart.com/app/diagrams/my-test-document-id
```

The benefits are:
  - It's more RESTful
  - It's much easier for users to just click the link and edit it on MermaidChart.com (assuming they have the correct permissions)
  - It makes it easier to see if a diagram has been synced to MermaidChart.com, or an on-premises Mermaid Chart instance.

The downsides are:
  - Parsing will be a bit more complicated
  - It may look a bit messy, since then we're adding more text to the diagram.

File: packages/cli/README.md
Comment (aloisklink) [inline packages/cli/README.md:15]: We can't test this until we've published this package to NPM, but it should work!

File: packages/cli/README.md
Comment (sidharthv96) [inline packages/cli/README.md:39]: Can we use OAuth before releasing?

File: packages/cli/src/commander.ts
Comment (sidharthv96) [inline packages/cli/src/commander.ts:238]: Can't we import it from the json file? That's how we do in mermaid.

File: packages/cli/src/methods.ts
Comment (sidharthv96) [inline packages/cli/src/methods.ts:14]: nit: Consistency

```suggestion
  selectedProjectID?: string;
```

**Review by sidharthv96 — APPROVED** [review]

File: packages/cli/src/commander.ts
Comment (aloisklink) [reply packages/cli/src/commander.ts]: Fixed by https://github.com/Mermaid-Chart/plugins/pull/12/commits/480df29378136f2d3636c78fe03d9c00fac1f717

I think in a previous version of this library, I had the file in `src/index.ts`, but I compiled the output into `./index.js`. So `require("../package.json");` would have had to have switched to `require("./package.json");`. But now I'm using a `dist/` folder, so the import path to `package.json` is the same!

File: packages/cli/README.md
Comment (aloisklink) [reply packages/cli/README.md:39]: IMO, since it's not a mandatory feature that's block the v0.1.0 release, I think it can wait for a future minor release (e.g. v0.1.1)!

But yep, we should definitely add it to make it easier for users to sign-in!

File: packages/cli/src/methods.ts
Comment (aloisklink) [reply packages/cli/src/methods.ts:14]: :+1:

I'll look into fixing all of these before we make the first release, but after #22 is merged, to avoid causing merge conflicts.

File: packages/cli/test/fixtures/connected-diagram.mmd
Comment (aloisklink) [reply packages/cli/test/fixtures/connected-diagram.mmd:3]: Will be fixed by #20!

**Pull request merged**
